homeassistant/components/foscam/camera.py

```diff
@@ -80,25 +89,26 @@ async def async_setup_entry(
         "async_perform_ptz_preset",
     )
 
-    camera = FoscamCamera(
-        config_entry.data[CONF_HOST],
-        config_entry.data[CONF_PORT],
-        config_entry.data[CONF_USERNAME],
-        config_entry.data[CONF_PASSWORD],
-        verbose=False,
-    )
+    coordinator: FoscamCoordinator = hass.data[DOMAIN][config_entry.entry_id][
+        "coordinator"
+    ]
 
-    async_add_entities([HassFoscamCamera(camera, config_entry)])
+    async_add_entities([HassFoscamCamera(coordinator, config_entry)])
 
 
-class HassFoscamCamera(Camera):
+class HassFoscamCamera(CoordinatorEntity[FoscamCoordinator], Camera):
     """An implementation of a Foscam IP camera."""
 
-    def __init__(self, camera: FoscamCamera, config_entry: ConfigEntry) -> None:
+    def __init__(
+        self,
+        coordinator: FoscamCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
         """Initialize a Foscam camera."""
-        super().__init__()
+        super().__init__(coordinator)
+        Camera.__init__(self)
 
-        self._foscam_session = camera
+        self._foscam_session = coordinator.get_session()
```

Comment:
```suggestion
        self._foscam_session = coordinator._session
```
Perhaps change `_session_ to `session` to avoid complains from pylint